extension-localization/src/main/java/com/mapbox/maps/extension/localization/Localization.kt

```diff
@@ -34,9 +39,16 @@ internal fun setMapLanguage(locale: Locale, style: StyleInterface, layerIds: Lis
               if (BuildConfig.DEBUG) {
                 Logger.i(TAG, "Localize layer id: ${it.layerId}")
               }
-              val language = if (sourceIsStreetsV8(style, source)) getLanguageNameV8(locale)
-              else getLanguageNameV7(locale)
-              convertExpression(language, it, textFieldExpression)
+
+              if (sourceIsStreetsV8(style, source)) {
+                convertedLocale = getLanguageNameV8(locale)
+              }
+
+              if (sourceIsStreetsV7(style, source)) {
```

Comment:
Could it be v7 and v8 at the same time? If no - maybe use if-else or when instead?